src/java/org/apache/cassandra/index/sai/disk/v1/IndexWriterConfig.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.v1;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.lucene.index.VectorSimilarityFunction;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.SAI_VECTOR_SEARCH_MAX_TOP_K;
+
+/**
+ * Per-index config for storage-attached index writers.
+ */
+public class IndexWriterConfig
+{
+    public static final String MAXIMUM_NODE_CONNECTIONS = "maximum_node_connections";
```

Comment:
I think the constants related to the same property could be grouped together for easier reading, as in:
```java
public static final String MAXIMUM_NODE_CONNECTIONS = "maximum_node_connections";
public static final int MAXIMUM_MAXIMUM_NODE_CONNECTIONS = 512;
public static final int DEFAULT_MAXIMUM_NODE_CONNECTIONS = 16;

public static final String CONSTRUCTION_BEAM_WIDTH = "construction_beam_width";
public static final int MAXIMUM_CONSTRUCTION_BEAM_WIDTH = 3200;
public static final int DEFAULT_CONSTRUCTION_BEAM_WIDTH = 100;
```